.gitlab-ci.yml

```diff
@@ -0,0 +1,84 @@
+stages:
+  - lint
+  - test
+
+
+variables:                         &default-vars
+  GIT_STRATEGY:                    fetch
+  GIT_DEPTH:                       100
+  CARGO_INCREMENTAL:               0
+  CARGO_TARGET_DIR:                "/ci-cache/${CI_PROJECT_NAME}/targets/${CI_COMMIT_REF_NAME}/${CI_JOB_NAME}"
+  CI_IMAGE:                        "paritytech/ci-linux:production"
+
+.rust-info-script:                 &rust-info-script
+  - rustup show
+  - cargo --version
+  - rustup +nightly show
+  - cargo +nightly --version
+  - bash --version
+  - sccache -s
+
+.docker-env:                       &docker-env
+  image:                           "${CI_IMAGE}"
+  before_script:
+    - *rust-info-script
+    - ./scripts/ci/pre_cache.sh
+    - sccache -s
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - linux-docker
+
+.kubernetes-env:                   &kubernetes-env
+  image:                           "${CI_IMAGE}"
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - kubernetes-parity-build
+
+.collect-artifacts:                &collect-artifacts
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}"
+    when:                          on_success
+    expire_in:                     28 days
+    paths:
+      - ./artifacts/
+
+#### stage:                        lint
+
+fmt:
+  stage:                           lint
+  <<:                              *kubernetes-env
+  script:
+    - cargo fmt --all -- --check
+
+clippy:
+  stage:                           lint
+  <<:                              *docker-env
+  script:
+    - cargo +nightly clippy --all --verbose
+    - RUSTDOCFLAGS="--deny broken_intra_doc_links" cargo doc --verbose --workspace --no-deps --document-private-items
+
+#### stage:                       test
+
+test:
```

Comment:
Further, do you mind copy-pasting [this](https://github.com/paritytech/jsonrpsee/blob/master/.github/workflows/ci.yml#L49-#L123) here too?

The reason why is that some feature might be leaked from the workspace to individual crates that we need check... however we should probably run with `--all-features` after removing some mutually exclusive feature flags..